server/app/boards.go

```diff
@@ -250,20 +250,67 @@ func (a *App) CreateBoard(board *model.Board, userID string, addMember bool) (*m
 		if addMember {
 			a.wsAdapter.BroadcastMemberChange(newBoard.TeamID, newBoard.ID, member)
 		}
+		if newBoard.ChannelID != "" {
+			members, err := a.GetMembersForBoard(board.ID)
+			if err != nil {
+				a.logger.Error("Unable to get the board members", mlog.Err(err))
+			}
+			for _, member := range members {
+				a.wsAdapter.BroadcastMemberChange(newBoard.TeamID, member.BoardID, member)
+			}
+		}
 		return nil
 	})
 
 	return newBoard, nil
 }
 
 func (a *App) PatchBoard(patch *model.BoardPatch, boardID, userID string) (*model.Board, error) {
+	var oldMembers []*model.BoardMember
+	if patch.ChannelID != nil && *patch.ChannelID == "" {
+		var err error
+		oldMembers, err = a.GetMembersForBoard(boardID)
+		if err != nil {
+			a.logger.Error("Unable to get the board members", mlog.Err(err))
+		}
+	}
 	updatedBoard, err := a.store.PatchBoard(boardID, patch, userID)
 	if err != nil {
 		return nil, err
 	}
 
 	a.blockChangeNotifier.Enqueue(func() error {
 		a.wsAdapter.BroadcastBoardChange(updatedBoard.TeamID, updatedBoard)
+		if patch.ChannelID != nil && *patch.ChannelID != "" {
+			members, err := a.GetMembersForBoard(updatedBoard.ID)
+			if err != nil {
+				a.logger.Error("Unable to get the board members", mlog.Err(err))
+			}
+			for _, member := range members {
+				a.wsAdapter.BroadcastMemberChange(updatedBoard.TeamID, member.BoardID, member)
+			}
+		} else if patch.ChannelID != nil && *patch.ChannelID == "" {
+			members, err := a.GetMembersForBoard(updatedBoard.ID)
```

Comment:
I'm not sure I'm understanding this piece of logic. If the non-synthetic membership always takes precedence over the synthetic one, how is it that I'm going to get a synthetic membership on `oldMembers` and a non-synthetic one after removing the channel link? If the non-synthetic membership was there, I would have fetched it on the `oldMembers`, so there are no chances for the `syntheticOnly` condition to be `true`, right?

Reply:
I see what you mean, really good catch, I'll fix that.

Reply:
Done